app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/triage/MultipleDisposalsTriageController.scala

```diff
@@ -314,39 +323,132 @@ class MultipleDisposalsTriageController @Inject() (
     }
   }
 
+  def whenWereContractsExchanged(): Action[AnyContent] = authenticatedActionWithSessionData.async { implicit request =>
+    withMultipleDisposalTriageAnswers(request) {
+      case (_, _, answers) =>
+        val taxYearExchanged = answers.fold(_.taxYearAfter6April2020, _ => Some(true))
+        val form             = taxYearExchanged.fold(taxYearExchangedForm)(taxYearExchangedForm.fill)
+        Ok(taxYearExchangedPage(form, routes.MultipleDisposalsTriageController.wereAllPropertiesResidential()))
+    }
+  }
+
+  def whenWereContractsExchangedSubmit(): Action[AnyContent] = authenticatedActionWithSessionData.async {
+    implicit request =>
+      withMultipleDisposalTriageAnswers(request) {
+        case (_, state, answers) =>
+          taxYearExchangedForm
+            .bindFromRequest()
+            .fold(
+              formWithErrors =>
+                BadRequest(
+                  taxYearExchangedPage(
+                    formWithErrors,
+                    routes.MultipleDisposalsTriageController.wereAllPropertiesResidential()
+                  )
+                ), { taxYearAfter6April2020 =>
+                if (answers.fold(_.taxYearAfter6April2020, _ => Some(true)).contains(taxYearAfter6April2020)) {
+                  Redirect(routes.MultipleDisposalsTriageController.checkYourAnswers())
+                } else {
+
+                  val result =
+                    for {
+                      taxYear <- if (taxYearAfter6April2020) taxYearService.taxYear(LocalDateUtils.today())
+                                else EitherT.pure(None)
+                      answers <- EitherT.fromEither[Future](
+                                  updateTaxYearToAnswers(taxYearAfter6April2020, taxYear, answers)
+                                )
+                      newState = state.copy(newReturnTriageAnswers = Left(answers))
+                      _ <- EitherT(
+                            updateSession(sessionStore, request)(_.copy(journeyStatus = Some(newState)))
+                          )
+                    } yield ()
+
+                  result.fold(
+                    { e =>
+                      logger.warn("Could not find tax year or update session", e)
+                      errorHandler.errorResult()
+                    },
+                    _ => Redirect(routes.MultipleDisposalsTriageController.checkYourAnswers())
+                  )
+
+                }
+              }
+            )
+      }
+  }
+
+  private def updateTaxYearToAnswers(
+    taxYearAfter6April2020: Boolean,
+    taxYear: Option[TaxYear],
+    answers: MultipleDisposalsTriageAnswers
+  ): Either[Error, MultipleDisposalsTriageAnswers] =
+    taxYear match {
+      case None if taxYearAfter6April2020 =>
+        Left(Error("You cannot use this service"))
```

Comment:
Something more descriptive like `Could not find tax year` would be better here - this is what we'll see in the logs in kibana.